dashboard/app/models/course.rb

```diff
@@ -188,7 +188,7 @@ def update_scripts(new_scripts, alternate_scripts = nil)
       CourseScript.where(course: self, script: script).destroy_all
     end
     # Reload model so that default_course_scripts is up to date
-    reload
+    transaction {reload}
```

Comment:
This operation caused an issue in read-write splitting mode, since it sends a SELECT query for the specific row immediately after it was initially written. Wrapping this kind of operation in a `transaction` block locks the query to the writer.